deployment/onnxruntime/main.cpp

```diff
@@ -0,0 +1,244 @@
+#include <iostream>
+#include <fstream>
+#include <codecvt>
+#include <opencv2/opencv.hpp>
+#include <onnxruntime_cxx_api.h>
+#include "cmdline.h"
+
+struct Detection
+{
+    cv::Rect box;
+    float conf{};
+    int classId{};
+};
+
+namespace utils
+{
+    std::wstring charToWstring(const char* str)
+    {
+        typedef std::codecvt_utf8<wchar_t> convert_type;
+        std::wstring_convert<convert_type, wchar_t> converter;
+
+        return converter.from_bytes(str);
+    }
+
+    std::vector<std::string> loadNames(const std::string& path)
+    {
+        // load class names
+        std::vector<std::string> classNames;
+        std::ifstream infile(path);
+        if (infile.good())
+        {
+            std::string line;
+            while (getline (infile, line))
+            {
+                classNames.emplace_back(line);
+            }
+            infile.close();
+        }
+        else
+        {
+            std::cerr << "ERROR: Failed to access class name path: " << path << std::endl;
+        }
+
+        return classNames;
+    }
+
+    void visualizeDetection(cv::Mat& image, std::vector<Detection>& detections, const std::vector<std::string>& classNames)
+    {
+        for (const Detection& detection : detections)
+        {
+            cv::rectangle(image, detection.box, cv::Scalar(229, 160, 21), 2);
+
+            int x = detection.box.x;
+            int y = detection.box.y;
+
+            int conf = (int)(detection.conf * 100);
+            int classId = detection.classId;
+            std::string label = classNames[classId] + " 0." + std::to_string(conf);
+
+            int baseline = 0;
+            cv::Size size = cv::getTextSize(label, cv::FONT_ITALIC, 0.8, 2, &baseline);
+            cv::rectangle(image, cv::Point(x, y - 25), cv::Point(x + size.width, y), cv::Scalar(229, 160, 21), -1);
+
+            cv::putText(image, label, cv::Point(x, y - 3), cv::FONT_ITALIC, 0.8, cv::Scalar(255, 255, 255), 2);
+        }
+    }
+}
+
+class Yolov5Detector
```

Comment:
Nit: To keep the naming scheme, maybe we could change `Yolov5Detector` to `YOLOv5Detector` here and below.